### DISCUSSION TITLE
how to know what version to download

### DISCUSSION THREAD

**Discussion by avnav0** [containers/podman#18016] [Q&A]
i'm trying to use ansible to install podman, and i want to specify an exact version.

when i install without doing that and run `apt-cache show podman | grep Version`, i get:

`Version: 3.0.1+dfsg1-3+deb11u1`

i'm wondering where i can find the full string for the latest version of 4.3.0.

this is a debian 11, so maybe it's the case that 3.0.1 is the latest stable version?  But i'd like to know if theres a page i could go to to find how a full string for ansible.

Thanks!

**Comment by vrothberg** [comment]
Thanks for reaching out, @avnav88.

I am not entirely sure I fully understand your issue but if you want to look for packages in Debian, the Debian package search is pretty helpful.  Podman 4.4.0, for instance, is in experimental: https://packages.debian.org/search?suite=experimental&arch=any&searchon=names&keywords=podman

**Reply by avnav0** [reply]
hey thank you for this!  I'm not sure what this string format is called: `4.4.0+ds1-1`

but that's what i was hoping to get.

**Reply by afbjorklund** [reply]
You should be able to use the version, but you also need to cache the packages for when Debian updates it...

https://packages.debian.org/search?keywords=podman&searchon=names&exact=1&suite=all&section=all

If you hardcode ("pin") the version, then it will start to fail when there is a package update with a new revision.

For now, I think what you have is the latest available for "bullseye" (11)

**Reply by avnav0** [reply]
you're saying if i pin it like so: `sudo -S apt-get -y install podman=3.0.1+dfsg1-3+deb11u1`

then it will fail when there's an update with a new version, unless i cache it somewhere.  do they just delete older version from the repos?  i was under the impression that it's recommended to hardcode the versions for dependency sake...

appreciate the insights btw

**Reply by afbjorklund** [reply]
Better to ask Debian, but that is my impression... Old versions might be archived somewhere*, but not in the apt search path ?

\* https://snapshot.debian.org/
